src/ColorSchemeToggle/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { useContext, type FC, type ReactElement } from "react";
+import ColorContext, {
+  type ColorContextValue,
+  type ColorScheme,
+} from "../ColorContext";
+import { nextSchemePreference } from "../ColorProvider";
+import { IoMoonSharp, IoSunnySharp } from "react-icons/io5";
+
+import "./index.scss";
+
+export interface ColorSchemeToggleProps {}
+
+const ColorSchemeToggle: FC<ColorSchemeToggleProps> = ({}) => {
+  let colorSchemeCtx = useContext(ColorContext);
+
+  return (
+    <button
+      className="color-scheme-toggle--button"
+      aria-label="Toggle light/dark/auto mode"
+      onClick={() => {
+        colorSchemeCtx.setPreference(
+          nextSchemePreference(colorSchemeCtx.preference),
+        );
+      }}
+    >
+      <div
+        className={`color-scheme-toggle--icon color-scheme-toggle--${colorSchemeCtx.scheme}`}
+      >
+        {iconForContext(colorSchemeCtx)}
+      </div>
+    </button>
+  );
+};
+
+function iconForContext(context: ColorContextValue): ReactElement {
+  switch (context.preference) {
+    case "auto":
+      return <AutoIcon colorScheme={context.scheme} />;
+    case "light":
+      return <IoSunnySharp height="auto" width="auto" />;
+
+    case "dark":
+      return <IoMoonSharp height="auto" width="auto" />;
+  }
+}
+
+interface AutoIconProps {
+  readonly colorScheme: ColorScheme;
+}
+
+const AutoIcon: FC<AutoIconProps> = ({ colorScheme }) => (
```

Comment:
This is a standalone component. Move it to `AutoIcon.tsx` as a sibling of this file. (It doesn't need to be in the top level or be exported by the index file.)